src/connection_string.ts

```diff
@@ -860,6 +878,33 @@ export const OPTIONS = {
   promoteValues: {
     type: 'boolean'
   },
+  proxyHost: {
+    target: 'proxyOptions',
+    transform({ values: [value], options }): ProxyOptions {
+      return { ...options.proxyOptions, host: String(value) };
+    }
+  } as OptionDescriptor,
+  proxyOptions: {
+    type: 'record'
+  },
+  proxyPassword: {
+    target: 'proxyOptions',
+    transform({ values: [value], options }): ProxyOptions {
+      return { host: '', ...options.proxyOptions, password: String(value) || undefined };
+    }
+  } as OptionDescriptor,
```

Comment:
If we do flatten these options on to the MongoClientOptions interface we shouldn't need to cast here anymore